### PR TITLE
tso: fix bugs to make split test case to pass

### PR DESCRIPTION
**Pull request by binshi-bing**
### What problem does this PR solve?

Issue Number: Ref #6232

### What is changed and how does it work?

```commit-message
fix bugs to make split test case to pass
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: tests/integrations/mcs/tso/keyspace_group_manager_test.go [hunk 1]
```diff
@@ -188,64 +186,6 @@ func (suite *tsoKeyspaceGroupManagerTestSuite) requestTSO(
 	return tam.HandleRequest(tsopkg.GlobalDCLocation, count)
 }
 
-func (suite *tsoKeyspaceGroupManagerTestSuite) TestTSOKeyspaceGroupSplitElection() {
```
Comment (lhy1024) [inline tests/integrations/mcs/tso/keyspace_group_manager_test.go:191]: why remove it?

File: pkg/tso/keyspace_group_manager.go [hunk 2]
```diff
@@ -131,8 +131,10 @@ func (s *state) getKeyspaceGroupMetaWithCheck(
 	// The keyspace doesn't belong to this keyspace group, we should check if it belongs to any other
 	// keyspace groups, and return the correct keyspace group meta to the client.
 	if kgid, ok := s.keyspaceLookupTable[keyspaceID]; ok {
-		return s.ams[kgid], s.kgs[kgid], kgid,
-			genNotServedErr(errs.ErrGetAllocatorManager, keyspaceGroupID)
+		if s.ams[kgid] != nil {
+			return s.ams[kgid], s.kgs[kgid], kgid, nil
+		}
+		return nil, s.kgs[kgid], kgid, genNotServedErr(errs.ErrGetAllocatorManager, keyspaceGroupID)
```
Comment (lhy1024) [inline pkg/tso/keyspace_group_manager.go:137]: when s.ams[kgid] is nil, can s.kgs[kgid] be nil at the same time?

Comment (binshi-bing) [reply pkg/tso/keyspace_group_manager.go:137]: no. s.kgs stores all keyspace groups; s.ams[kgid] only stores allocator managers of the keyspace groups assigned to this server/pod

**Review by lhy1024 — APPROVED** [review]

**Pull request merged**